doc/cephadm: document the new per-fsid cephadm conf location (#46369)
ceph/ceph: doc/cephadm/operations.rst

CEPHADM_CHECK_KERNEL_VERSION
~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The OS kernel version (maj.min) is checked for consistency across the hosts.

> The kernel version of the majority of the hosts is used as the basis for

Reviewer: Does the command analyze the results and identify anomalies, or are we saying that the *operator/admin* can use the results that way?

Author: Not sure about this part, I've just removed the spurious extra space.

Client keyrings and configs
===========================
Cephadm can distribute copies of the ``ceph.conf`` file and client keyring
files to hosts. Starting from versions 16.2.10 (Pacific) and 17.2.1 (Quincy),
in addition to the default location ``/etc/ceph/`` cephadm also stores config
and keyring files in the ``/var/lib/ceph/<fsid>/config`` directory. It is usually
a good idea to store a copy of the config and ``client.admin`` keyring on any host
used to administer the cluster via the CLI. By default, cephadm does this for any
nodes that have the ``_admin`` label (which normally includes the bootstrap host).

.. note:: the new configuration location ``/var/lib/ceph/<fsid>/config`` is used by

>   cephadm only. Ceph daemons will still use files on ``/etc/ceph/``.

Reviewer 2: can we add to this note why we also store a copy in `/var/lib/ceph/<fsid>/config`? Mainly, that it helps cephadm infer what config to use since it being in the fsid dir guarantees it's associated with this cluster?

Reviewer: Agreed, is the order in https://github.com/ceph/ceph/pull/46369#discussion_r879682390 documented elsewhere?

I can see people shooting themselves in the foot because they don't know where the canonical copy is, that will overwrite `/etc/ceph` files that their automation populates.

Author: @anthonyeleven right now is only documented on the cephadm shell section since it only affects this part (where cephadm shell loads the config). As stated in the note: this doesn't affect the configuration used by ceph daemons. 

To make it clear from where the config is taken, cephadm (as in the following example) shows the location when starting the shell. So the user normally should be aware from where the config is loaded:

```
[root@ceph-node-0 ~]# cephadm shell
Inferring fsid d87d598a-ebd2-11ec-b9b0-525400581dfe
Inferring config /var/lib/ceph/d87d598a-ebd2-11ec-b9b0-525400581dfe/mon.ceph-node-0/config
Using ceph image with id '08b9d8a1fe1b' and tag 'custom-v0.14' created on 2022-06-03 14:48:17 +0000 UTC
registry.hub.docker.com/rkachach/ceph@sha256:9933fba86fe39e492d04b6551c79e129855a003580629650fda17142b5b43907

```

Anyway, please suggest any additional suitable location to put this doc into in case you feel it would be helpful for end users.

Client keyrings and configs
===========================
Starting from versions 16.2.10 (Pacific) and 17.2.1 (Quincy) in addition to
the default location ``/etc/ceph/`` cephadm also stores config and keyring

> files in the ``/var/lib/ceph/<fsid>/config`` directory.

Reviewer 2: I feel like this paragraph fits better after the `Cephadm can distribute copies of . . .` paragraph than before it

Client keyrings and configs
===========================
By default, starting from versions 16.2.10 and 17.2.1 in addition to

> the default location ``/etc/ceph/`` ceph also stores its configuration

Reviewer 2: ```suggestion
the default location ``/etc/ceph/`` cephadm also stores its config and keyring
```
I feel like we should specify here this is a cephadm feature. The rest of ceph is still unaware of this config in /var/lib/ceph/<fsid> and would only check /etc/ceph (if it checks for the file at all)

Reviewer: Suggest also including a bit about why this is being done and how the new per-fsid copy will be used differently.

Reviewer: Groovy.  I'm thinking especially about installations that have existing automation to maintain `ceph.conf` that might run afoul of this.  I suspect that it might be akin to the way that in a previous life we used `ceph-deploy` from an admin node, and had to carefully keep `ceph.conf` in sync there with the copies in `/etc/ceph` on the daemon nodes.

Reviewer: Ack.  So it's possible then to manage more than one cluster from a given `cephadm` node?  That mirrors the situation in the abovementioned previous life, where initially we had one admin node per DC with potentially multiple clusters, and would have to cd into a cluster-specific subdirectory before running `ceph-deploy`.  This was awkward and prone to error, so we switched to a dedicated `ceph-deploy` node per cluster.

Author: Not sure about `ceph-deploy` (as I'm not familiar with the tool) but with `cephadm` things are different and you can specify which cluster you want to manage/operate (by specifying the fsid). In case `cephadm` detects multiple cluster on the host it will force the user to specify which of them he wants to manage. AFAIK there's no risk to mess other clusters running in the same host because each cluster has its own daemons and all the operations are performed from the `cephadm shell` which is runs everything per fsid and by using the corresponding cluster admin key.

> files on ``/var/lib/<fsid>/config`` directory.

Reviewer 2: ```suggestion
files in the ``/var/lib/ceph/<fsid>/config`` directory.
```

Author: Fixed

  - build a list of target hosts based on the specified placement spec (see
    :ref:`orchestrator-cli-placement-spec`)
  - store a copy of the ``/etc/ceph/ceph.conf`` file on the specified host(s)

>   - store a copy of ``ceph.conf`` file on ``/var/lib/<fsid>/config/ceph.conf`` on the specified host(s)

Reviewer 2: ```suggestion
  - store a copy of the ``ceph.conf`` file at ``/var/lib/ceph/<fsid>/config/ceph.conf`` on the specified host(s)
```

>   - store a copy of ``ceph.client.admin.keyring`` file on ``/var/lib/<fsid>/config/ceph.client.admin.keyring`` on the specified host(s)

Reviewer 2: ```suggestion
  - store a copy of the ``ceph.client.admin.keyring`` file at ``/var/lib/ceph/<fsid>/config/ceph.client.admin.keyring`` on the specified host(s)
```

Author: Fixed
